### PR TITLE
CLI to download multiple files

### PR DESCRIPTION
**Pull request by jsh2134**
Builds off of https://github.com/solvebio/solvebio-python/pull/325

```
 $ solvebio download --help
usage: solvebio download [-h] [--version] [--api-host API_HOST]
                         [--api-key API_KEY] [--access-token ACCESS_TOKEN]
                         [--dry-run]
                         full_path local_path

positional arguments:
  local_path            The path to the local directory where to download
                               files.
  full_path      The full path to the files on SolveBio. Supports Unix
                        style globs in order to download multiple files. Note:
                        Downloads are not recursive.
SolveBio Options:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  --api-host API_HOST   Override the default SolveBio API host
  --api-key API_KEY     Manually provide a SolveBio API key
  --access-token ACCESS_TOKEN
                        Manually provide a SolveBio OAuth2 access token
  --dry-run             Dry run mode will not download any files or create any
                        folders.
```

### CODE REVIEW

**Comment by davecap** [comment]
Would it make more sense to have the full path be a required argument to the command?

`solvebio download remote_path local_path`

**Comment by jsh2134** [comment]
I agree with that. Went ahead and made the change. I left the arg name as full_path to make sure its obvious that it is the full_path (or full_path glob) to the object. We could also change it to `remote_full_path`  if you think that is the best name

`solvebio download full_path local_path`

`solvebio download remote_full_path local_path`

**Comment by davecap** [comment]
What happens if you pass a folder name as the full path?

**Comment by jsh2134** [comment]
> What happens if you pass a folder name as the full path?

Ya what do you want to happen? It will download nothing, as it won't find any files. I tired to mitigate this case by [adding a useful error message here](https://github.com/solvebio/solvebio-python/pull/328/files#diff-6fcda87d3c582de192a4643a9d078c28R490)

We don't have any file path glob detection (we could? and if it is a valid full path, use it explicitly?)

**Comment by davecap** [comment]
It also doesn't work recursively yet right? So it won't create folders found in the remote path?

File: solvebio/cli/data.py [hunk 1]
```diff
@@ -460,6 +460,46 @@ def import_file(args):
...
+        print("Creating local download folder {}".format(local_folder_path))
+        if not dry_run:
+            if not os.path.exists(local_folder_path):
+                os.makedirs(local_folder_path)
+
+    # API will determine depth based on number of "/" in the glob
+    # Add */** to match in any vault (recursive)
+    files = Object.all(glob=full_path, limit=1000, object_type='file')
+    if not files:
+        print("No file(s) found at --full-path {}\nIf aattempting to download "
```
Comment (davecap) [inline solvebio/cli/data.py:490]: typo `aatempting`

**Comment by jsh2134** [comment]
> It also doesn't work recursively yet right? So it won't create folders found in the remote path?

Correct, it currently only downloads a items at a single depth. It will only create folders locally if the local_path is a dir

Here is pseudo-output
```
solvebio download vault:/path/to/files/*.txt my/local/path

  Creating my/local/path/ if not exists
  Downloading vault:/path/to/files/file1.txt to my/local/path/file1.txt
  Downloading vault:/path/to/files/file2.txt to my/local/path/file2.txt
```

**Comment by davecap** [comment]
ok!

**Pull request merged**